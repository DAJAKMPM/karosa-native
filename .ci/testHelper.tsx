import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
