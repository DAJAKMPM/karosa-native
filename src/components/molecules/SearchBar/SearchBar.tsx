/**
 *
 * SearchBar
 * @format
 *
 */

import React, { FC } from "react";
import { Platform } from "react-native";
import { SearchBar as RnSearchBar } from "react-native-elements";

import type { PropsType } from "./types";
import SearchBarStyles from "./styles";

const SearchBar: FC<PropsType> = (props) => {
  const {
    backgroundColor = "primary",
    size = "lg",
    autoCorrect,
    autoFocus,
    onChangeText,
    onCancel,
    onFocus,
    onTouchStart,
    placeholder,
    value,
  } = props;

  const {
    inputContainer: inputContainerStyle,
    mainContainer: mainContainerStyle,
  } = SearchBarStyles(size, backgroundColor);

  return (
    <RnSearchBar
      platform={Platform.OS === "android" ? "android" : "ios"}
      placeholder={placeholder}
      inputContainerStyle={inputContainerStyle}
      autoCorrect={autoCorrect}
      containerStyle={mainContainerStyle}
      returnKeyType="search"
      accessibilityRole="search"
      autoFocus={autoFocus}
      onTouchStart={onTouchStart}
      onCancel={onCancel}
      onFocus={onFocus}
      onChangeText={onChangeText}
      value={value}
    />
  );
};

export default SearchBar;
