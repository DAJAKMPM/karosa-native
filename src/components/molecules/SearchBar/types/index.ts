/**
 *
 * @format
 *
 */

import { GestureResponderEvent } from "react-native";

export type SizeType = "sm" | "md" | "lg";

export type BackgroundColorType = "primary" | "secondary";

export type PropsType = {
  value?: string;
  placeholder?: string;
  autoCorrect?: boolean;
  size?: SizeType;
  autoFocus?: boolean;
  backgroundColor?: BackgroundColorType;
  onChangeText?: (text: string) => void;
  onFocus?(): void;
  onCancel?(): void;
  onTouchStart?: (event: GestureResponderEvent) => void;
};
