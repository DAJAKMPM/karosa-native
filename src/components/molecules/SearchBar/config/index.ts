/**
 *
 * @format
 *
 */

import { theme } from "@app/styles";

import { BackgroundColorType, SizeType } from "../types";

// will use dimensions for getting size, will rework
const defaultSize: Record<SizeType, number> = {
  sm: 222,
  md: 225,
  lg: 252,
};

export const getWidth = (size: SizeType): number => {
  switch (size) {
    case "md":
      return defaultSize.sm;
    case "lg":
      return defaultSize.lg;
    default:
      return defaultSize.sm;
  }
};

export const getBackgroundColor = (backgroundColor: BackgroundColorType) => {
  switch (backgroundColor) {
    case "secondary":
      return theme.colors.light10;
    default:
      return theme.colors.white;
  }
};
