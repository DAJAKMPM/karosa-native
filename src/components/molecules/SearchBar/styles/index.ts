/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

import { BackgroundColorType, SizeType } from "../types";
import { getWidth, getBackgroundColor } from "../config";

const SearchBarStyles = (
  size: SizeType,
  backgroundColor: BackgroundColorType
) => {
  return StyleSheet.create({
    inputContainer: {
      height: 42,
      width: getWidth(size),
      backgroundColor: getBackgroundColor(backgroundColor),
    },
    mainContainer: {
      backgroundColor: theme.colors.transparent,
      borderTopColor: theme.colors.transparent,
      borderBottomColor: theme.colors.transparent,
    },
  });
};

export default SearchBarStyles;
