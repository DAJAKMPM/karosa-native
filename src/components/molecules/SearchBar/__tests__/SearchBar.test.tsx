import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import SearchBar from "../SearchBar";

describe("SearchBar Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<SearchBar />);
    expect(toJSON()).toMatchSnapshot();
  });
});
