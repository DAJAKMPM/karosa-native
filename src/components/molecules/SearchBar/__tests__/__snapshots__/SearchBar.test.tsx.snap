// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "flexGrow": 1,
        "height": 40,
        "overflow": "hidden",
        "paddingBottom": 0,
        "paddingTop": 0,
        "position": "relative",
        "top": -5,
        "width": 675,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "borderRadius": 9,
            "flexDirection": "row",
            "marginLeft": 8,
            "marginRight": 8,
            "minHeight": 36,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <TextInput
          accessibilityRole="search"
          allowFontScaling={true}
          editable={true}
          lightTheme={false}
          onBlur={[Function]}
          onCancel={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="#7d7d7d"
          platform="ios"
          rejectResponderTermination={true}
          returnKeyType="search"
          round={false}
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "marginLeft": 6,
              "minHeight": 40,
              "overflow": "hidden",
            }
          }
          testID="searchInput"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginRight": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "height": 0,
            "margin": 0,
            "padding": 0,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "opacity": 0,
          "position": "absolute",
          "right": null,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
